public/src/components/CheckboxButton.jsx

```diff
@@ -3,35 +3,21 @@ import { PropTypes } from 'prop-types';
 import styled, { css } from 'styled-components';
 
 const StyledLabel = styled.label`
+  ${({ theme }) => css`
+    ${theme.text.contentText}
+    ${theme.button.defaultStyling}
+    &:focus-within {
+      box-shadow: ${theme.button.boxShadowSizes} rgba(219, 112, 147, 0.7);
+    }
+    & path {
+      transition: fill ${theme.button.transitionTiming} ease-in;
+      fill: palevioletred;
+    }
+    transition: background-color ${theme.button.transitionTiming} ease-in, color ${theme.button.transitionTiming} ease-in, box-shadow ${theme.button.transitionTiming} ease-in;
```

Comment:
So there's a neat concept in styled components called mixins.

```
// args: [[property: String, transition: String]]
const transition = (args) => css`
  ${({ theme }) => css`
       transition: ${args.map(([property, transition]) => `${property} ${theme.button.transitionTiming} ${transition}`).join(', ')};
    `
`;
```

Then this becomes:

```jsx
${transition(['background-color', 'ease-in'], ['color', 'ease-in'], ['box-shadow', 'ease-in'])}
```

Reply:
Nice! Implemented.